hail/python/hail/expr/functions.py

```diff
@@ -3638,16 +3660,20 @@ def sorted(collection,
     :class:`.ArrayExpression`
         Sorted array.
     """
-    ascending = ~reverse
+
+    rev = Env.get_uid()
+    def comp(l, r):
+        return (hl.case()
+                .when(hl.is_missing(l), False)
+                .when(hl.is_missing(r), True)
+                .when(reverse, hl._compare(l, r) > 0)
+                .default(hl._compare(l, r) > 0))
```

Comment:
I think one of these last two cases needs to be different.

Reply:
(and why didn't this trigger a test failure?)

Reply:
argh, thanks for catching that!